pkgs/servers/pufferpanel/default.nix

```diff
@@ -2,23 +2,54 @@
 , buildGoModule
 , fetchFromGitHub
 , makeWrapper
-, pkgs
 , stdenv
 , fetchzip
-, jdk
-, nodejs
+, fetchpatch
 , pathDeps ? [ ]
 }:
 
 buildGoModule rec {
   pname = "pufferpanel";
-  version = "2.2.0";
+  version = "2.6.6";
+
+  patches = [
+    # Bump go-sqlite3 version.
+    # See https://github.com/PufferPanel/PufferPanel/pull/1240
+    (fetchpatch {
+      url = "https://github.com/PufferPanel/PufferPanel/pull/1240/commits/3065dca2d9b05a56789971ccf0f43a7079a390b8.patch";
+      hash = "sha256-ygMrhJoba8swoRBBii7BEiLihqOebLUtSH7os7W3s+k=";
+    })
+
+    # Fix errors in tests.
+    # See https://github.com/PufferPanel/PufferPanel/pull/1241
+    (fetchpatch {
+      url = "https://github.com/PufferPanel/PufferPanel/pull/1241/commits/ffd21bce4bff3040c8e3e783e5b4779222e7a3a5.patch";
+      hash = "sha256-BzGfcWhzRrCHKkAhWf0uvXiiiutWqthn/ed7bN2hR8U=";
+    })
+
+    # Seems to be an anti-feature. Startup is the only place where user/group is
+    # hardcoded and checked.
+    #
+    # There is no technical reason PufferPanel cannot run as a different user,
+    # especially for simple commands like `pufferpanel version`.
+    ./disable-group-checks.patch
+
+    # Some tests do not have network requests stubbed :(
+    ./skip-network-tests.patch
+  ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=github.com/pufferpanel/pufferpanel/v2.Hash=${src.rev}"
+    "-X=github.com/pufferpanel/pufferpanel/v2.Version=${version}"
+  ];
 
   src = fetchFromGitHub {
     owner = "pufferpanel";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "1ifig8ckjlg47wj0lfk4q941dan7llb1i5l76akcpjq726b2j8lh";
+    rev = "cd0016247711ce7197e0a141decc895119239ed5"; # "v${version}"
```

Comment:
What happened to this? Did they stop cutting tags?

Reply:
Nope, but I’m using `src.rev` in `ldflags`. Now that I think about it, it’s probably better to just put `commitish` as a derivation argument and use that instead of `src.rev`. What do you think?

Reply:
For reference, PufferPanel uses Git SHA in `ldflags` to set version information.
https://github.com/PufferPanel/PufferPanel/blob/0c8cca79c92b7bf42793a2259e999f2b1a9c69ec/.github/workflows/build.yml#L272

So we mimic this behavior.
https://github.com/NixOS/nixpkgs/blob/bcce2fe6365ccb66831e6259c4f4bff097da4894/pkgs/servers/pufferpanel/default.nix#L41-L46

Reply:
It looks like [elsewhere](https://github.com/ckiee/nixpkgs/blob/b5eda8810115633af15ad39228384e25c123bf51/pkgs/applications/networking/cluster/kyverno/default.nix#L17) (there's a few other matches, not a ton of them though) we stick the version in the commit hole, or in this one case, [set it to a dummy value](https://github.com/ckiee/nixpkgs/blob/b5eda8810115633af15ad39228384e25c123bf51/pkgs/applications/networking/cluster/hubble/default.nix#L19). 

I think one of those would be better, maybe also stick on a `-nixpkgs` to give any fellow future debuggers a hint, if you want.

Reply:
Done, set `…/v2.Hash=none` and added `-nixpkgs` suffix to version.

Reply:
Another `…` user :D